corehq/apps/hqwebapp/models.py

```diff
@@ -81,6 +87,32 @@ def __new__(cls, category, action, label=None):
         return super(GaTracker, cls).__new__(cls, category, action, label)
 
 
+class UUIDGeneratorMixin(object):
+    """
+    Automatically generates uuids on __init__ if not generated yet.
+
+    To use: Add this mixin to your model as the left-most class being inherited from
+    and list all field names in UUIDS_TO_GENERATE to generate uuids for.
+
+    NOTE: Where possible, a UUIDField should be used instead of this mixin. But
+    this is needed in cases where migrating a char field to a UUIDField is
+    not possible because some of the existing uuids don't match the
+    UUIDField format constraints.
+    """
```

Comment:
wow nice :skull: ! 

just so I understand - the purpose of this field is so that `couch_id` is not empty on certain SQL models? It just feels a little strange that we would set a random ID that is meaningless. could we not just remove the constraints on that field if that's the issue? (sorry if I'm misunderstanding how this is used).

(maybe worth updating the doc string with the practical use cases for this mixin so that's clear in the future)